asserts/crypto.go

```diff
@@ -178,3 +230,55 @@ func parsePublicKey(pubKey []byte) (PublicKey, error) {
 	}
 	return WrapPublicKey(pubk), nil
 }
+
+// EncodePublicKey serializes a public key, typically for embedding in an assertion.
+func EncodePublicKey(pubKey PublicKey) ([]byte, error) {
+	return encodeOpenpgp(pubKey, "public key")
+}
+
+// PrivateKey is a cryptographic private/public key pair.
+type PrivateKey interface {
+	// PublicKey returns the public part of the pair.
+	PublicKey() PublicKey
+}
+
+type openpgpPrivateKey struct {
+	privk *packet.PrivateKey
+}
+
+func (opgPrivK openpgpPrivateKey) PublicKey() PublicKey {
+	return WrapPublicKey(&opgPrivK.privk.PublicKey)
+}
+
+func (opgPrivK openpgpPrivateKey) openpgpSerialize(w io.Writer) error {
```

Comment:
This makes me wonder if what we want isn't actually `keySerialize(w io.Writer)` and `keyFormat() string`. This turns the `openpgpSerializer` interface into `keySerializer`, and `openpgpEncode` above into `keyEncode`, reducing the amount of "openpgp" we have in the code.